tensilelite/Tensile/CustomKernels/Custom_Cijk_Alik_Bljk_HHS_BH_UserArgs_MT128x16x128_MI16x16x1_SN_K1_MIWT2_1_triple_buffer.s

```diff
@@ -0,0 +1,3569 @@
+s_lshl_b64 s[74:75], s[74:75], s[sgprGSULog2BpeD]  // scale by bpe
+s_add_u32 s[sgprSrdD+0], s[sgprSrdD+0], s74        // add lo to SRD
+s_addc_u32 s[sgprSrdD+1], s[sgprSrdD+1], s75       // add hi to SRD
+
+s_cmp_eq_u32 s[sgprGSU], 1                         // GSU == 1 ?
+s_cbranch_scc1 label_GSU_2                         // branch if GSU == 1
+// GSU Output Buffer offset: Free0 + (Free1-1)*StrideC1J + (Free2-1)*StrideCK * GSUIdx * bpe%s
+s_mul_hi_u32 s75, s[sgprSizesFree+0], s[sgprGSUSumIdx] // Free0
+s_mul_i32 s74, s[sgprSizesFree+0], s[sgprGSUSumIdx] // Free0
+s_sub_u32 s73, s[sgprSizesFree+1], 1               // Free1
+s_mul_i32 s73, s73, s[sgprGSUSumIdx]               // Free1
+s_mul_hi_u32 s76, s73, s[sgprStrideC1J]            // Free1
+s_mul_i32 s73, s73, s[sgprStrideC1J]               // Free1
+s_add_u32 s74, s74, s73                            // Free1
+s_addc_u32 s75, s75, s76                           // Free1
+s_sub_u32 s73, s[sgprSizesFree+2], 1               // Free2
+s_mul_i32 s73, s73, s[sgprGSUSumIdx]               // Free2
+s_mul_hi_u32 s76, s73, s[sgprStrideCK]             // Free2
+s_mul_i32 s73, s73, s[sgprStrideCK]                // Free2
+s_add_u32 s74, s74, s73                            // Free2
+s_addc_u32 s75, s75, s76                           // Free2
+s_lshl_b64 s[74:75], s[74:75], 2                   // scale by bpe
+s_add_u32 s[sgprSrdD+0], s[sgprSrdD+0], s74        // add lo GSU offset to SRD
+s_addc_u32 s[sgprSrdD+1], s[sgprSrdD+1], s75       // add hi GSU offset to SRD
+label_GSU_2:
+.set sgprGSULog2BpeC, UNDEF
+
+/* initC: remove ValuC vgpr buffer [0...0) from pool */
+
+/* initC: remove acc vgpr buffer [0...8) from pool */
+
+/* initC: remove ValuA/B vgpr buffer [0...48) from pool */
+v_accvgpr_write acc0, 0x0                          // initC
+v_accvgpr_write acc1, 0x0                          // initC
+v_accvgpr_write acc2, 0x0                          // initC
+v_accvgpr_write acc3, 0x0                          // initC
+v_accvgpr_write acc4, 0x0                          // initC
+v_accvgpr_write acc5, 0x0                          // initC
+v_accvgpr_write acc6, 0x0                          // initC
+v_accvgpr_write acc7, 0x0                          // initC
+s_cmp_eq_u32 s[sgprLoopCounterL], 0                // at last iteration?
+
+/* after InitC, skip to end of prefetch last iter if numIter==0 */
+s_cbranch_scc0 label_NoBranch_LMI4T0VTC74OFN9W_0   // Only branch on scc1
+s_getpc_b64 s[74:75]                               // addr of next instr
+s_add_i32 s76, label_PrefetchGlobalLastIterEnd, 0x4 // target branch offset
+s_add_u32 s74, s74, s76                            // add target branch offset
+s_addc_u32 s75, s75, 0                             // add high and carry
+s_setpc_b64 s[74:75]                               // branch to label_PrefetchGlobalLastIterEnd
+label_NoBranch_LMI4T0VTC74OFN9W_0:
+s_waitcnt vmcnt(0)                                 // 8wait for global read
```

Comment:
We actually do not need to wait 2nd PGR here.
I am not sure which is better
1) add condition to use different vmcnt for 1 iteration and more than 1 iteration
2) move 2nd PGR after ds_write.

Reply:
Updated with option #1, thanks.